files/static/index.js

```diff
@@ -0,0 +1,340 @@
+  },
+
+  setTheme: function(theme) {
+    this.clearTheme();
+    document.body.classList.add(theme)
+  },
+
+  currentDomain: function() {
+    var domain = $("#domain").value;
+    // Check for pasted URLs beginning with http:// or https://
+    if (domain.startsWith("http://")) {
+      domain = domain.slice("http://".length)
+    }
+    if (domain.startsWith("https://")) {
+      domain = domain.slice("https://".length)
+    }
+    return domain;
+  },
+
+  clearResults: function() {
+    $("#result").classList.add("hidden");
+  },
+
+  showResults: function(domain, checkDomainIssues, domainStatus) {
+    if (checkDomainIssues.errors.length === 0) {
+      if (checkDomainIssues.warnings.length === 0) {
+        $("#summary").textContent = "" + domain + " satisfies all requirements for preloading!";
+        this.setTheme("theme-green");
+      } else {
+        $("#summary").textContent = "" + domain + " has " + (checkDomainIssues.warnings.length == 1 ? "a warning" : "warnings") + ", but satisfies all requirements for preloading.";
+        this.setTheme("theme-yellow");
+      }
+    } else {
+        $("#summary").textContent = "" + domain + " has errors and cannot be preloaded.";
+        this.setTheme("theme-red");
+    }
+
+    switch (domainStatus.status) {
+    case "unknown":
+      if (checkDomainIssues.errors.length === 0) {
+        $("#status").textContent = "" + domain + " is not yet preloaded, and may be submitted to the preload list."
+      } else {
+        $("#status").textContent = "" + domain + " is not preloaded."
+      }
+      break;
+    case "pending":
+      $("#status").textContent = "" + domain + " is pending submission to the preload list."
+      break;
+    case "preloaded":
+      $("#status").textContent = "" + domain + " is currently preloaded."
+      break;
+    case "rejected":
+      // TODO: do something useful if domainStatus.message is not present.
+      if (domainStatus.message) {
+        $("#status").textContent = "" + domain + " was rejected from the preload list for the following reason: " + domainStatus.message;
+      } else {
+        $("#status").textContent = "" + domain + " was rejected from the preload list.";
+      }
+      if (checkDomainIssues.errors.length === 0) {
+        $("#status").textContent += " It may be submitted again."
+      }
+      break;
+    case "removed":
+      if (checkDomainIssues.warnings.length === 0) {
+        $("#status").textContent = "" + domain + " was previously on the preload list, but has been removed. It may be submitted again.";
+      } else {
+        $("#status").textContent = "" + domain + " was previously on the preload list, but has been removed.";
+      }
+      break;
+    default:
+      $("#status").textContent = "Cannot determine preload status.";
+    }
+
+    function createIssueElement(issue, bullet, className) {
+      var el = document.createElement("div");
```

Comment:
Since there's not so much of it, I'm going taking the DevTools approach of pure JS. I've defined `Element.prototype.createChild` to make things easier to read: https://github.com/chromium/hstspreload.appspot.com/blob/master/files/static/js/view.js#L127